solutions/cses-1090.mdx

```diff
@@ -81,4 +82,53 @@ int main(){
 
 </CPPSection>
 
+<JavaSection>
+
+<Warning>
+Java will TLE on test case #10.
```

Comment:
dang java so slow its fine for usaco but codeforces and other websites nah cuz no extra time :(

Reply:
try using fast io?

Reply:
isn't kattio already fast io? it uses buffered reader which is the fastest way I think?

Reply:
or is this something that i don't know about

Reply:
he is referring to https://usaco.guide/general/fast-io/#method-5---inputstream i believe

Reply:
rip FastIO still TLE's on tc 10

Reply:
hmmm I see java solutions that pass with scanner lol https://cses.fi/problemset/hack/1090/entry/1990829/

Reply:
Can you send a pic? For some reason the link just shows a blank screen

Reply:
i'll send the code:

<details>

<summary>
Code
</summary>

```java
import java.util.Arrays;
import java.util.Scanner;
 
public class CSES1090 {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		
		while (input.hasNext()) {
			int n = input.nextInt();
			int x = input.nextInt();
			input.nextLine();
			String[] pStr = input.nextLine().split(" ");
			Integer[] p = new Integer[n];
			
			for (int i = 0; i < p.length; ++i) {
				p[i] = Integer.decode(pStr[i]);
			}
			
			Arrays.sort(p);
			
			int ans = 0;
			for (int i = 0, j = p.length - 1; i <= j; --j) {
				if (i != j && x - p[j] >= p[i]) {
					++i;
				}
				++ans;
			}
			
			System.out.println(ans);
		}
	}
}
```

</details>

Reply:
this is just bad the solutions to problems should be on the algorithm and not on how you use IO

Reply:
This uses the same logic as my code, I guess the way they read in the array is faster?

Reply:
btw I fixed the TLE issue so it now passes the all test cases